### ISSUE TITLE
Allow HTTP requests with status 0 to be retried

### ISSUE THREAD

**Issue by evanweible-wf** [Workiva/w_transport#173]
When an XHR request is made from the browser, there are some special case failures that may occur. The browser may not be able to complete the request because there is no network connection, or it may refuse to complete the request as a security measure (example: server did not return access-control headers, thus rejecting the cross-origin request). In the network tab, you'll likely see that the request returned with a status code of 0. This is not an official HTTP status code, but rather a special code used by browsers to indicate the type of failure to the developer.

In code, a response will most likely not be received at all. Instead, an error event will be triggered on the XHR instance. With our current implementation, this results in the error being raised (and then wrapped in a `RequestException`) and a null `response`. Additionally, our "can retry" logic treats a null `response` as a non-retryable failure. I'm not sure this should be the case.

On one hand, we don't know exactly why a request failed when it fails like this, so it may be useless to retry it (CORS). On the other hand, one of the more common scenarios in which this would happen is a lack of network connection (offline mode or intermittent connectivity), which would benefit from retrying.

**tl;dr:** we should consider allowing retries for failed requests where the failure is unknown.
